lib/ask/runtime/session.ex

```diff
@@ -269,21 +293,40 @@ defmodule Ask.Runtime.Session do
       schedule
       |> Schedule.at_end_time(next_available_date_time)
 
-    channel_state =
-      runtime_channel
-      |> Channel.setup(session.respondent, token, next_available_date_time, today_end_time)
-      |> handle_setup_response()
+    channel = session.current_mode.channel
 
-    %{session | channel_state: channel_state, token: token}
+    :ok =
+      ChannelBroker.setup(
+        channel.id,
+        channel.type,
+        runtime_channel,
+        session.respondent,
+        token,
+        next_available_date_time,
+        today_end_time
+      )
+
+    %{session | token: token}
```

Comment:
Despite its generic name, the `channel_state` was used only to keep the call_id on IVR mode. Because this `setup()` call is now async here we don't have that value anymore. Now the `call_id` is kept in the ChannelBroker state [here](https://github.com/instedd/surveda/blob/e520169e209128b30dea32f6315337b51d241f30/lib/ask/runtime/channel_broker.ex#L428-L450) in the `active_contacts` map. When `ChannelBroker.has_queued_message?`, `ChannelBroker.cancel_message` and `ChannelBroker.message_expired?` are called it takes the verbioce_call_id from the state and calls [these](https://github.com/instedd/surveda/blob/01cd890c1f6031a0644044523a2311b43eea84b1/lib/ask/runtime/channel.ex#L6-L8) Channel methods including the `call_id`. Maybe the `channel_state` param of these functions should be renamed or a deeper refactor should take place. I didn't want to go further just because so many changes scared me.